packages/ui/src/views/repo/components/branch-selector/branch-selector.tsx

```diff
@@ -21,16 +23,20 @@ export const BranchSelector: FC<BranchSelectorProps> = ({
   buttonSize = 'default',
   selectedBranch,
   onSelectBranch,
-  isBranchOnly = false
+  isBranchOnly = false,
+  searchQuery = '',
+  setSearchQuery
 }) => {
   const { selectedBranchTag, branchList, tagList, repoId, spaceId } = useRepoBranchesStore()
 
   const isTag = selectedBranchTag
     ? tagList?.some(tag => tag.name === selectedBranchTag.name && tag.sha === selectedBranchTag.sha)
     : false
 
+  const [open, setOpen] = useState(false)
+
   return (
-    <DropdownMenu>
+    <DropdownMenu open={open} onOpenChange={setOpen}>
```

Comment:
why does this need to be a controlled component? where are we using `setOpen`? i dont see it being passed anywhere

Reply:
yes not required anymore, initially I added to trap focus but handled from within the component, removing this